libraries/zookeeper_service.rb

```diff
@@ -86,6 +86,7 @@ class ZookeeperService < Chef::Provider
       include PoiseService::ServiceMixin
 
       def action_enable
+        new_resource.notifies(:restart, new_resource, :delayed)
```

Comment:
Just because this looks new to me and haven't seen this before. Could you just briefly explain why this is done here and not directly in the provider in the recipe?